src/record/kv.rs

```diff
@@ -0,0 +1,290 @@
+// Copyright 2024 FastLabs Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::borrow::Cow;
+use std::fmt;
+use std::slice;
+
+use value_bag::OwnedValueBag;
+use value_bag::ValueBag;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub(crate) enum KeyInner<'a> {
```

Comment:
Just for interests sake, I have a `Cow`-like string type I carry around with me that avoids the cost of matching on variants for each access: https://github.com/emit-rs/emit/blob/main/core/src/str.rs#L28-L34

Whether it turns out to be better or not depends on how frequently you're looking at those strings.

Reply:
@KodrAus Curious, what's the reason for not storing `Box<str>` directly for `StrOwner::Box` but a pointer instead?